app/functoria_app.ml

```diff
@@ -417,81 +435,113 @@ module Make (P: S) = struct
     ()
 
   let clean_main i jobs =
-    Engine.clean i jobs >>| fun () ->
-    Cmd.remove (Info.root i / "main.ml")
+    Engine.clean i jobs >>= fun () ->
+    Bos.OS.File.delete Fpath.(v "main.ml")
 
   let configure i jobs =
-    Log.info "%a %s" Log.blue "Using configuration:"  (get_config_file ());
-    Log.info "opam: %a" (Info.opam ?name:None) i ;
-    Cmd.in_dir (Info.root i) (fun () -> configure_main i jobs)
+    Log.info (fun m -> m "Using configuration: %s" (get_config_file ()));
+    Log.info (fun m -> m "opam: %a" (Info.opam ?name:None) i);
+    Log.info (fun m -> m "within: %s" (Info.root i));
+    match
+      Bos.OS.Dir.with_current (Fpath.v (Info.root i))
+        (fun () -> configure_main i jobs) ()
+    with
+    | Ok a -> a
+    | Error _ -> R.error_msg "failed to change directory for configure"
+
+  let build i jobs =
+    Log.info (fun m -> m "Building: %s" (get_config_file ()));
+    match
+      Bos.OS.Dir.with_current (Fpath.v (Info.root i))
+        (fun () -> Engine.build i jobs) ()
+    with
+    | Ok a -> a
+    | Error _ -> R.error_msg "failed to change directory for build"
 
   let clean i (_init, job) =
-    Log.info "%a %s" Log.blue "Clean:"  (get_config_file ());
+    Log.info (fun m -> m "Cleaning: %s" (get_config_file ()));
     let root = Info.root i in
-    Cmd.in_dir root (fun () ->
-        clean_main i job >>= fun () ->
-        Cmd.run "rm -rf %s/_build" root >>= fun () ->
-        Cmd.run "rm -rf log %s/main.native.o %s/main.native %s/*~" root
-          root root
-      )
+    match
+      Bos.OS.Dir.with_current (Fpath.v root)
+        (fun () ->
+           clean_main i job >>= fun () ->
+           Bos.OS.Dir.delete ~recurse:true (Fpath.v "_build") >>= fun () ->
+           Bos.OS.File.delete (Fpath.v "log") >>= fun () ->
+           Bos.OS.File.delete (Fpath.v "main.native.o") >>= fun () ->
+           Bos.OS.File.delete (Fpath.v "main.native")) ()
```

Comment:
was the same before https://github.com/mirage/functoria/blob/d7a20157078e1f1e9b2680ce24031e44d19faaa3/app/functoria_app.ml#L488

since this PR is already pretty big, I'd leave the `main.native` question to a subsequent PR.